ESCPOS_NET/Printers/BasePrinter.cs

```diff
@@ -1,51 +1,71 @@
 ﻿using System;
-using System.Collections;
-using System.Collections.Concurrent;
+using System.Collections.Concurrent;
+using System.Diagnostics;
 using System.IO;
 using System.Linq;
-using System.Text;
 using System.Threading;
 using System.Threading.Tasks;
-using System.Timers;
+using System.Timers;
 
 namespace ESCPOS_NET
 {
     public abstract partial class BasePrinter : IDisposable
     {
-        public PrinterStatus Status = null;
+    	private bool disposed = false;
+        private volatile bool _isMonitoring;
```

Comment:
this is cool, never heard of volatile keyword.  was there a detectable issue based on this or did you just know that this was appropriate?

Reply:
Because it is a read operation it may not get the actual value, it may be cached using the volatile keyword means it won't use the cached version. 
A better explantion can be found here: [Volatile keyword C#](https://www.infoworld.com/article/3229360/how-to-use-the-volatile-keyword-in-c.html) the quote below is from this article.

> C# all memory writes are volatile irrespective of whether you are writing data to a volatile or a non-volatile object. However, the ambiguity happens when you are reading data. When you are reading data that is non-volatile, the executing thread may or may not always get the latest value. If the object is volatile, the thread always gets the most up-to-date value.